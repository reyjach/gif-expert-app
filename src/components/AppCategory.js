import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AppCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const hadleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const hadleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length >2){
      setCategories( cat => [inputValue,...cat]);
      setInputValue('');
    }
    
  }

  return (
      <form onSubmit={ hadleSubmit }>
        <input 
          type='text'
          value={inputValue}
          onChange={ hadleInputChange }
        />
      </form>
  )
}

AppCategory.propTypes = {
  setCategories:PropTypes.func.isRequired
}
