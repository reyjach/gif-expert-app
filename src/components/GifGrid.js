import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data:images, loading }  = useFetchGifs( category );

    return (
        <>
            <h3 className='card animate__animated animate__bounceOutRight'>{category}</h3>
            {loading && <p className='card animate__animated animate__fadeOut'>Loading</p> }
            <div className='card-grid'>             
                { 
                    images.map(img => (
                    <GifGridItem key={img.id} {...img}/>
                    )) 
                }
            </div>
        </>
        
    )
}
GifGrid.propTypes = {
    category:PropTypes.string.isRequired
  }