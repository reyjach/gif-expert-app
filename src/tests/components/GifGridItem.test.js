import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";


describe('Probar el componente GifGridItem', () => { 

    const title = 'Un titulo';

    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow( <GifGridItem title={title} url={url}/>);


    test('Debe de mostrar el componente correcto', () => { 

        expect(wrapper).toMatchSnapshot()

    });

    test('Debe de tener un parro con el title', () => { 

        const parrafo = wrapper.find('p').text().trim();

        expect(parrafo).toBe(title);

    });

    test('Debe de tener una imagen al url y un alt de los props', () => {

        const img = wrapper.find('img').props();

        expect(img.src).toBe(url);
        expect(img.alt).toBe(title);

    });

    test('Debe de tener la clase animate__bounceIn', () => { 

        const div = wrapper.find('div').props().className;

        expect(div.includes('animate__bounceIn')).toBe(true);
     })
 })