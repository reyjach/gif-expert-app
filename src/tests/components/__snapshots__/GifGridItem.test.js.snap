// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el componente GifGridItem Debe de mostrar el componente correcto 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
